File: 101.symmetric-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
	public boolean isSymmetric(TreeNode root) {
		return root == null||isSymmetricHelper(root.left,root.right);        
	}

	public boolean isSymmetricHelper(TreeNode left,TreeNode right){
		if(left == null && right == null)
			return true;
		if(left == null || right == null)
			return false;
		if(left.val == right.val) 
			return isSymmetricHelper(left.left,right.right)&&isSymmetricHelper(left.right,right.left);
		return false;
	}
}

File: 101.symmetric-tree.js
/*
 * @lc app=leetcode id=101 lang=javascript
 *
 * [101] Symmetric Tree
 */
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSymmetric = function (root) {
  if (!root) {
    return true;
  }
  return isSymmetricHelper(root.left, root.right);
};


isSymmetricHelper = (left, right) => {
  if (!left && !right) {
    return true;
  }

  if (!left || !right) {
    return false;
  }

  if (left.val == right.val) {
    return isSymmetricHelper(left.left, right.right) && isSymmetricHelper(left.right, right.left)
  } else {
    return false;
  }

};


